lib/bundler/shared_helpers.rb

```diff
@@ -336,12 +336,23 @@ def clean_load_path
       loaded_gem_paths = Bundler.rubygems.loaded_gem_paths
 
       $LOAD_PATH.reject! do |p|
-        next if File.expand_path(p).start_with?(bundler_lib)
+        next if resolve_path(p).start_with?(bundler_lib)
         loaded_gem_paths.delete(p)
       end
       $LOAD_PATH.uniq!
     end
 
+    def resolve_path(path)
+      expanded = File.expand_path(path)
+      return expanded unless File.respond_to?(:realpath)
+
+      while File.exist?(expanded) && File.realpath(expanded) != expanded
+        expanded = File.realpath(expanded)
```

Comment:
AFAIK File.realpath() resolves all symlinks it can, so a single call should be enough, unlike `readlink(1)`.

Reply:
Thanks @ojab for making a PR to fix this :)